New version: GModelFit v0.5.0

JuliaRegistrator opened the pull request:
- Registering package: GModelFit
- Repository: https://github.com/gcalderone/GModelFit.jl
- Created by: @gcalderone
- Version: v0.5.0
- Commit: ca71a0ce8970d5879caee8c61cc29e02e357fd41
- Git reference: HEAD
- Description: A data-driven model fitting framework
- Release notes:
<!-- BEGIN RELEASE NOTES -->
`````
GModelFit v0.5.0 now supports all solvers from NonlinearSolve, as well as forward mode automatic differentiation via ForwardDiff. A signficant refactor was needed to accomodate the above functionalities, as well as to improve performance, resulting in a number of breaking changes:

* Signature of the `evaluate!` functions for component evaluation has changed (this may break your code if you implemented custom components, check `evaluate!` documentation);

* The `prepare!` function is now supposed to be used just for pre-computing quantities, while the length of the evaluation vector is now provided by the `result_length` function;

* The solver is now passed as third argument to the `fit` and `fit!` functions (rather than being specified using the `solver=` keyword);

* The `FitStats` structure is has been replaced by the `FitSummary` one.
`````
<!-- END RELEASE NOTES -->

gcalderone commented:
@JuliaRegistrator register

Release notes:

## Breaking changes

- GModelFit now supports all solvers from NonlinearSolve, as well as forward mode automatic differentiation via ForwardDiff;
- Signature of the `evaluate!` functions for component evaluation has changed (this may break your code if you implemented custom components, check `evaluate!` documentation);
- The `prepare!` function is now supposed to be used just for pre-computing quantities, while the length of the evaluation vector is now provided by the `result_length` function;
- The solver is now passed as third argument to the `fit` and `fit!` functions (rather than being specified using the `solver=` keyword);
- The `FitStats` structure is has been replaced by the `FitSummary` one.

gcalderone commented:
Hi,
sorry, I am a bit lost here....

Where should I add the release notes?
Both the README.md and ChangeLog.md files mention the breaking changes.  What else should I do?

Thanks!

JuliaHubRegistrator committed:
New version: GModelFit v0.5.0

UUID: ac42e6ba-c0bf-4fcf-827d-deea44b16255
Repo: https://github.com/gcalderone/GModelFit.jl.git
Tree: 87174a3d72db21dccf9738034b5294412508a591

Registrator tree SHA: d5716b7a540e5fbc43640c2fff2906fe50e9525a
